google/cloud/opentelemetry/integration_tests/monitoring_exporter_integration_test.cc

```diff
@@ -0,0 +1,147 @@
+namespace cloud {
+namespace otel {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::testing::IsEmpty;
+using ::testing::Not;
+namespace metrics_api = ::opentelemetry::metrics;
+namespace metrics_sdk = ::opentelemetry::sdk::metrics;
+namespace sc = opentelemetry::sdk::resource::SemanticConventions;
+
+auto constexpr kMeterName =
+    "gl-cpp/testing/monitoring_exporter_integration_test";
+
+std::string RandomId() {
+  auto generator = internal::MakeDefaultPRNG();
+  return internal::Sample(generator, 8, "0123456789");
+}
+
+// If this code is unfamiliar, see the "simple" metrics example:
+// https://github.com/open-telemetry/opentelemetry-cpp/tree/2d077f8ec5315e0979a236554c81f621eb61f5b3/examples/metrics_simple
+void InstallExporter(std::unique_ptr<metrics_sdk::PushMetricExporter> exporter,
+                     std::string const& run_id) {
+  // GCM requires that metrics be tied to a Monitored Resource. Instead of using
+  // a GCP Resource Detector, which requires that the integration test be run
+  // from GCP, we set attributes that we know will map to a `generic_node`.
+  auto resource = opentelemetry::sdk::resource::Resource::Create(
+      {{sc::kServiceNamespace, "gl-cpp"},
+       {sc::kServiceName, "monitoring_exporter_integration_test"},
+       {sc::kServiceInstanceId, run_id}});
+
+  // Initialize and set the global MeterProvider
+  metrics_sdk::PeriodicExportingMetricReaderOptions options;
+  options.export_interval_millis = std::chrono::milliseconds(5000);
+  options.export_timeout_millis = std::chrono::milliseconds(500);
+
+  // `PeriodicExportingMetricReaderFactory::Create(...)` was added in
+  // opentelemetry-cpp v1.10.0. We support >= 1.9.1.
+  std::unique_ptr<metrics_sdk::MetricReader> reader{
+      new metrics_sdk::PeriodicExportingMetricReader(std::move(exporter),
+                                                     options)};
+
+  // `MetricProviderFactory::Create(...)` was added in opentelemetry-cpp
+  // v1.10.0. We support >= 1.9.1.
+  std::unique_ptr<opentelemetry::metrics::MeterProvider> u_provider(
+      new metrics_sdk::MeterProvider(
+          std::make_unique<opentelemetry::sdk::metrics::ViewRegistry>(),
+          resource));
+  auto* p = static_cast<metrics_sdk::MeterProvider*>(u_provider.get());
+
+  p->AddMetricReader(std::move(reader));
+
+  std::shared_ptr<opentelemetry::metrics::MeterProvider> provider(
+      std::move(u_provider));
+  metrics_api::Provider::SetMeterProvider(provider);
+}
+
+void DoWork(std::string const& run_id) {
+  auto provider = metrics_api::Provider::GetMeterProvider();
+  opentelemetry::nostd::shared_ptr<metrics_api::Meter> meter =
+      provider->GetMeter(kMeterName);
+  auto double_counter = meter->CreateDoubleCounter(kMeterName);
+
+  // This takes 10s to run. That is unfortunate, but necessary because GCM has a
+  // minimum update period of 5s.
+  for (std::uint32_t i = 0; i < 20; ++i) {
+    double_counter->Add(i, {{"run_id", run_id}});
```

Comment:
The attribute is now redundant?

Reply:
You are right. We can achieve the same thing by filtering on the `resource.task_id` label. Removed. Thanks and sorry for the churn.